### PR TITLE
Add basic API documentation for /api/challenges.json

### LINKED ISSUE #22: Add API documentation
For http://paircolumbus.org/api/challenges.json

### PR DESCRIPTION
**Pull request by nickserv**
Fixes #22.

### CODE REVIEW

File: pages/api.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+layout: page
+title: API
+permalink: /api/
+---
+
+The Pair Columbus API is a public, read-only resource for information related to the group and its resources. For now, it just serves static JSON used to generate the [Challenges page](/challenges/).
+
+## [Challenges](/api/challenges.json)
+This endpoint returns all active Pair Columbus challenges in categories (generally based on difficultly or technologies used).
```
Comment (cartoloupe) [inline pages/api.md]: typo, "difficulty"

File: pages/api.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+### Request
+`GET http://paircolumbus.org/api/challenges.json`
```
Comment (cartoloupe) [inline pages/api.md:13]: I can get results from Postman, but not with `curl "http://paircolumbus.org/api/challenges.json"`... is this expected?

File: pages/api.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+description
+: One or more sentences describing the intention of the challenges in the category. Note that some descriptions contain [Markdown](http://daringfireball.net/projects/markdown/) content, so you may need to render them with a Markdown parser first.
+
+challenges
+: A list of strings of challenge names. Each challenge is named exactly after the name of its repository in the [Pair Columbus GitHub Organization](https://github.com/paircolumbus). To get the URL of a challenge's repository, prepend `https://github.com/paircolumbus/` to its name.
+
+#### Example
+This example is a subset of the real results, and has added whitespace for readability.
+
```
Comment (cartoloupe) [inline pages/api.md:32]: Consider adding the GET action to the example

**Comment by cartoloupe** [comment]
Awesome feature and docs! :+1:

File: pages/api.md (hunk 2 above)
Comment (nickserv) [inline pages/api.md:13]: Using curl, sometimes I get a 200 and sometimes I see this 302:

```
~ → curl --include http://paircolumbus.org/api/challenges.json
HTTP/1.1 302 Found
Connection: close
Pragma: no-cache
cache-control: no-cache
Location: /api/challenges.json

```

I also noticed that caching is disabled by default in Postman. Could this be a cached response?

File: pages/api.md (hunk 3 above)
Comment (nickserv) [inline pages/api.md:32]: I just remembered that GET is already in the Request section above, so I'll leave it out for now. I might reformat the docs later.

File: pages/api.md (hunk 2 above)
Comment (nickserv) [inline pages/api.md:13]: It seems like this is a side effect of how curl handles caching, since Postman disables caching by default.

**Pull request merged**